aten/src/ATen/ThreadLocalState.h

```diff
@@ -22,22 +45,23 @@ class TORCH_API ThreadLocalState {
   // according to the thread boundary specified
   static void setThreadLocalState(const ThreadLocalState& state);
 
+  // Some of the settings (e.g. profiler) may register
+  // getters and setters,
+  // WARNING: not thread safe
+  static void registerThreadLocalSetting(
```

Comment:
Is this going to be the general API for devs to register thread local settings that they want to propagate across threads? If so, it might be useful to add some documentation to this indicating how it can be used

Reply:
I didn't want to add this API, but as of now it is a necessity because of linking; I can add an extra doc